nuxeo-core/nuxeo-core-event/src/test/java/org/nuxeo/ecm/core/scheduler/EventJobFactoryTest.java

```diff
@@ -0,0 +1,76 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Munch
+ */
+package org.nuxeo.ecm.core.scheduler;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.EventListenerDescriptor;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import javax.inject.Inject;
+import java.lang.reflect.Field;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests {@link EventJobFactory} by contributing a job using {@link org.quartz.DisallowConcurrentExecution}.
+ *
+ * @since 10.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(SchedulerFeature.class) // starts Quartz with four threads to test concurrent executions
+@Deploy("org.nuxeo.ecm.core.event.test:test-jobfactory.xml") // jobs and listener
+public class EventJobFactoryTest {
+    @Inject
+    private EventService eventService;
+
+    private DummyScheduledJob eventListener;
+
+    @Before
+    public void setUp() throws Exception {
+        Field listenerField = EventListenerDescriptor.class.getDeclaredField("inLineListener");
+        listenerField.setAccessible(true);
+
+        // replacing the EventListener by a spy
+        EventListenerDescriptor eventDescriptor = eventService.getEventListener("testScheduler");
+        eventListener = (DummyScheduledJob) listenerField.get(eventDescriptor);
+        eventListener = spy(eventListener);
+        listenerField.set(eventDescriptor, eventListener);
+    }
+
+    @Test
+    public void test() throws InterruptedException {
+        // sleeping 5s, but the scheduler has already started and the jobs are already running, so the number of
+        // executions might vary
+        Thread.sleep(5000L);
+
+        // without DisallowConcurrentExecution the 2s job must be executed every second (even if executions overlap),
+        // meaning that the listener should be called at least four or five times
+        verify(eventListener, atLeast(4)).handleEvent("testSchedulerMultipleExecutions");
+
+        // with DisallowConcurrentExecution the 2s job cannot be executed every second and must wait for one execution
+        // to finish to run again, meaning that the listener cannot be called more than 3 times
+        verify(eventListener, atLeast(2)).handleEvent("testSchedulerSingleExecution");
+        verify(eventListener, atMost(3)).handleEvent("testSchedulerSingleExecution");
```

Comment:
Generally we don't like to have tests that sleep because it slows down the whole run — but for a scheduler test there's not many other ways to do testing.

In a perfect world these tests would be ok, however on slow machines we've seen that time-based tests are often not very reliable. I would prefer tests that do explicit waits and releases of the job based on `CountDownLatch`. If you see what that entails then please try to update the PR for that. Otherwise we'll push as is but will need to carefully watch for failures and if that's the case we'll have to disable the test.